app/components/DocumentExplorer.tsx

```diff
@@ -0,0 +1,372 @@
+import FuzzySearch from "fuzzy-search";
+import { includes, difference, concat, filter, flatten } from "lodash";
+import { observer } from "mobx-react";
+import { StarredIcon, DocumentIcon } from "outline-icons";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList as List } from "react-window";
+import styled, { useTheme } from "styled-components";
+import breakpoint from "styled-components-breakpoint";
+import { NavigationNode } from "@shared/types";
+import parseTitle from "@shared/utils/parseTitle";
+import DocumentExplorerNode from "~/components/DocumentExplorerNode";
+import DocumentExplorerSearchResult from "~/components/DocumentExplorerSearchResult";
+import Flex from "~/components/Flex";
+import CollectionIcon from "~/components/Icons/CollectionIcon";
+import EmojiIcon from "~/components/Icons/EmojiIcon";
+import { Outline } from "~/components/Input";
+import InputSearch from "~/components/InputSearch";
+import Text from "~/components/Text";
+import useCollectionTrees from "~/hooks/useCollectionTrees";
+import useMobile from "~/hooks/useMobile";
+import useStores from "~/hooks/useStores";
+import { isModKey } from "~/utils/keyboard";
+import { flattenTree, ancestors, descendants } from "~/utils/tree";
+
+type Props = {
+  actionOnItem: () => void;
```

Comment:
Need comments on these as it's not immediately clear what the difference between `actionOnItem` and `onSelectItem` is.

`actionOnItem` -> `onSubmit` perhaps?

Reply:
`onSubmit` is good, yes. Will add comments...